subfun/BIDS_Results.m

```diff
@@ -0,0 +1,142 @@
+matlabbatch = [];
+
+% loop trough the steps and more results to compute for each contrast
+% mentioned for each step
+for iStep = 1:length(opt.result.Steps)
+    
+    for iCon = 1:length(opt.result.Steps(iStep).Contrasts)
+        
+        % Depending on the level step we migh have to define a matlabbatch
+        % for each subject or just on for the whole group
+        switch opt.result.Steps(iStep).Level
+            
+            case 'run'
+                error('run level not implemented yet');
+                
+            case 'subject'
+                
+                for iGroup= 1:length(group)
+                    
+                    groupName = group(iGroup).name ;
+
+                    % For each subject
+                    for iSub = 1:group(iGroup).numSub   
+
+                        % Get the Subject ID
+                        subNumber = group(iGroup).subNumber{iSub} ;   
+                        
+                        % FFX Directory
+                        ffxDir = getFFXdir(subNumber, FFXSmoothing, opt, isMVPA);
+                        
+                        load(fullfile(ffxDir, 'SPM.mat'))
+                        
+                        % identify which contrast number actually has the name the user asked for 
+                        conNb = find ( strcmp({SPM.xCon.name}', opt.result.Steps(iStep).Contrasts(iCon).Name) );
+                        
+                        if isempty(conNb)
+                            sprintf('List of contrast in this SPM file')
+                            disp({SPM.xCon.name}')
+                            error('This SPM file %s does not contain a contrast named %s', ...
+                                fullfile(Dir, 'SPM.mat'), ...
+                                opt.result.Steps(1).Contrasts(iCon).Name)
+                        end
+                        
+                        matlabbatch = result_matlabbatch(matlabbatch, opt, iStep, iCon, ffxDir, conNb, subNumber, 1);
+                        
+                    end
+                end
+                
+            case 'dataset'
+
+                % Define the RFX folder name and create it in the derivatives
+                % directory
+                rfxDir = fullfile(opt.dataDir, '..', 'derivatives', 'SPM12_CPPL', ...
+                    ['RFX_', opt.taskName],...
+                    ['RFX_FunctSmooth', num2str(FFXSmoothing),...
+                    '_ConSmooth_', num2str(ConSmoothing)], ...
+                    opt.result.Steps(iStep).Contrasts(iCon).Name) ;
+                
+                load(fullfile(rfxDir, 'SPM.mat'))
+                
+                matlabbatch = result_matlabbatch(matlabbatch, opt, iStep, iCon, rfxDir, 1, 'group level', SPM.nscan);
+
+        end
+    end
+    
+
+end
+
+if ~isempty(matlabbatch)
+% save the matlabbatch
+save(fullfile(JOBS_dir, ...
+    'jobs_matlabbatch_SPM12_Results.mat'), ...
+    'matlabbatch')
+
+spm_jobman('run',matlabbatch)
+
+% move ps file
```

Comment:
Ideally we might want to rename each file that is created to make it easier to know which one is which when uploading to neurovault.